matter_server/common/models.py

```diff
@@ -106,6 +109,74 @@ class ServerDiagnostics:
     events: list[dict]
 
 
+class NodeType(Enum):
+    """Enum with Matter node types."""
+
+    END_DEVICE = "end_device"
+    SLEEPY_END_DEVICE = "sleepy_end_device"
+    ROUTING_END_DEVICE = "routing_end_device"
+    BRIDGE = "bridge"
+    UNKNOWN = "unknown"
+
+    @classmethod
+    def from_matter_routing_role(cls: Type, role: int) -> NodeType:
+        """Parse NetworkType from Matter RoutingRole integer."""
+        if role in (5, 6):
+            return NodeType.ROUTING_END_DEVICE
+        if role == 2:
+            return NodeType.SLEEPY_END_DEVICE
+        if role == 3:
+            return NodeType.END_DEVICE
+        return NodeType.UNKNOWN
+
+
+class NetworkType(Enum):
+    """Enum with Matter network types."""
+
+    THREAD = "thread"
+    WIFI = "wifi"
+    ETHERNET = "ethernet"
+    UNKNOWN = "unknown"
+
+    @classmethod
+    def from_matter_interface_type(cls: Type, interface: int) -> NetworkType:
+        """Parse NetworkType  from Matter InterfaceType integer."""
+        if interface == 1:
+            return NetworkType.WIFI
+        if interface == 2:
+            return NetworkType.ETHERNET
+        if interface == 4:
+            return NetworkType.THREAD
+        return NetworkType.UNKNOWN
+
+
+NodePingResult = dict[str, bool]
+
+
+@dataclass
+class ActiveFabric:
+    """Representation of a ActiveFabric message."""
+
+    fabric_index: int
+    vendor_id: int
+    fabric_id: int
+    name: str
+
+
+@dataclass
+class NodeDiagnostics:
```

Comment:
I don't agree as this is what makes it more useable/readable. A very small abstraction layer on top of those huge clusters. If you want the pure/raw data, just use read_attribute. This is all about creating a unified diagnostics object that we can easily use on the HA frontend.

Reply:
Note that we are already parsing some of the data into our own format (such as ip addresses)

Reply:
> I don't agree as this is what makes it more useable/readable. A very small abstraction layer on top of those huge clusters. If you want the pure/raw data, just use read_attribute. This is all about creating a unified diagnostics object that we can easily use on the HA frontend.

But if we start doing that here, why don't we do that for lights, vacuums and other things?

> So to give a more concrete answer: I am not conflating the WiFi and Thread diagnostics clusters. I'm just taking parts from it and model it into our own very small abstraction layer in a format useable for quick diagnostics overview of a node.

I agree, we need to simplify this for the frontend. And that is where the code belongs: Client side! We don't need to make our WS API more complex. We already do exchange data using  the Cluster objects, so why not stay consistent?

Reply:
I think you are overreacting here. I really do not get your issue here.

Reply:
As discussed privately; I'm are going to split this up into two parts. Ping will be server side in the custom modelled format. The diagnostics modeling will stay but client only so that the server can stay as simple/flat as possible